src/main/resources/Diagram/DiagramViewSheet.xml

```diff
@@ -319,6 +325,26 @@ require.config({
     diagramViewerDeferred.reject();
   });
 
+  // Return rawDiagram (non-XML) if is any diagram in URL
+  function getDiagramFromURL() {
+    if (window.location.hash !== '') {
+      // We need to exlude first 2 letters from hash (#R).
+      let decodedData = decodeURIComponent(window.location.hash.substring(2));
+      var graph = new Graph();
+      let rawDiagram = graph.decompress(decodedData);
+      return rawDiagram;
+    }
+    return null;
+  };
+
+  function getXMLDiagram($) {
```

Comment:
This function does two things: gets the XML and parses it. In order to make it more generic / reusable (what if later we would need only the XML) then I would split the two concerns:

* ``getDiagramXMLFromURL``
* ``getDiagramXML`` (calls ``getDiagramXMLFromURL``)
* and, if you want (but I don't think it's necessary because it's a one-liner), ``getDiagramXMLDocument``

Reply:
I refactored it, but I'm still not sure that's the change you wanted.